util/dvsim/NcLauncher.py

```diff
@@ -86,49 +85,49 @@ def _do_launch(self):
 
         self._dump_env_vars(exports)
 
+        # using os.open instead of fopen as this allows
+        # sharing of file descriptors across processes
+        fd = os.open(self.deploy.get_log_path(), os.O_WRONLY | os.O_CREAT)
+        os.set_inheritable(fd, True)
+        message = '[Executing]:\n{}\n\n'.format(self.deploy.cmd)
+        byte_message = message.encode('utf-8')
+        os.write(fd, byte_message)
+
         if self.deploy.sim_cfg.interactive:
             # Interactive: Set RUN_INTERACTIVE to 1
             exports['RUN_INTERACTIVE'] = '1'
-            interactive_flags = ['-I']
-            cmd_arr = self.get_submit_cmd(interactive_flags)
-            log.log(VERBOSE, '[Executing]:\n{}\n\n'.format(self.deploy.cmd))
+            cmd_arr = self.get_submit_cmd(['-I'])
+            buf_size = 1
+            std_out = subprocess.PIPE
+            std_err = subprocess.STDOUT
+        else:
+            cmd_arr = self.get_submit_cmd(['-wl'])
+            buf_size = 4096
```

Comment:
If 4k buffers aren't particularly important, is there a reason not to use `-1` (letting the system pick whatever it fancies)?

(I know this was in the previous code, but I'm wondering about it because the line is being touched :-D)

Reply:
done. added a comment